### ISSUE TITLE
`spago upgrade` should use a compatible compiler version if no sets for the current one are available

### ISSUE THREAD

**Issue by peterbecich** [purescript/spago#1191]
The latest package set 50.1.1 https://github.com/purescript/registry/blob/main/package-sets/50.1.1.json
specifies PureScript 0.15.14.

Can that be updated to solve this error from `spago upgrade`?
```
❌ No package set found for the current compiler version.
```

Thanks

**Comment by f-f** [comment]
Oh sorry, I was entirely convinced it was in spago in the first place. I moved it now.

**Comment by fsoikin** [comment]
From what I understand, the `Y >= Z` condition is already the case, as specified here: https://github.com/fsoikin/spago/blame/3c2b1ce92a5d9d1edbd236181eb821cabafd9c6a/src/Spago/Registry.purs#L295

And indeed, I was able to upgrade to the latest package set with compiler version lower than the current installed one:

```

 λ npx purs --version
0.15.16-3

 λ npx spago registry package-sets

+---------+------------+----------+
| VERSION | DATE       | COMPILER |
+---------+------------+----------+
| .... skipped lots ....          |
| 53.2.0  | 2024-07-01 | 0.15.15  |
| 53.2.1  | 2024-07-09 | 0.15.15  |
| 53.3.0  | 2024-07-10 | 0.15.15  |
| 53.3.1  | 2024-07-11 | 0.15.15  |
| 54.0.0  | 2024-07-12 | 0.15.15  |
| 55.0.0  | 2024-07-14 | 0.15.15  |
| 55.1.0  | 2024-07-18 | 0.15.15  |
| 55.2.0  | 2024-07-21 | 0.15.15  |
| 55.3.0  | 2024-07-24 | 0.15.15  |
+---------+------------+----------+

 λ npx spago upgrade

Reading Spago workspace configuration...

✅ Selecting package to build: spago-pg

Upgrading the package set to the latest version: 55.3.0

✅ Upgrade successful!
```

The only requirement remaining is the support for `--package-set` parameter.

Is that right?

**Comment by f-f** [comment]
> From what I understand, the Y >= Z condition is already the case

Ah yeah, it looks like I accidentally fixed this in #1228 😄 

Tangential thought: looking at your `.... skipped lots ....` makes me think that maybe the default for `spago registry package-sets` should be use to just show the latest package sets for each compiler (which now you need the `--latest` flag for), and have like a `--all` flag to show all of them instead

**Comment by fsoikin** [comment]
Yes, that makes sense, I was going to suggest it. But it doesn't sound like something blocking the milestone.
